leaflet-editable/leaflet-editable.d.ts

```diff
@@ -0,0 +1,256 @@
+// Project: https://github.com/yohanboniface/Leaflet.Editable
+// Definitions by: Dominic Alie <https://github.com/dalie>
+// Definitions: https://github.com/borisyankov/DefinitelyTyped
+
+/// <reference path="../leaflet/leaflet.d.ts" />
+
+declare module L {
+    /**
+     * Make geometries editable in Leaflet.
+     *
+     * This is not a plug and play UI, and will not. This is a minimal, lightweight, and fully extendable API to
+     * control editing of geometries. So you can easily build your own UI with your own needs and choices.
+     */
+    export interface EditableStatic {
+        new (map: Map, options: EditOptions): Editable;
+    }
+
+    /**
+     * Options to pass to L.Editable when instanciating.
+     */
+    export interface EditOptions {
+        /**
+         * Class to be used when creating a new Polyline.
+         */
+        polylineClass?: Object;
+
+        /**
+         * Class to be used when creating a new Polygon.
+         */
+        polygonClass?: Object;
+
+        /**
+         * Class to be used when creating a new Marker.
+         */
+        markerClass?: Object;
+
+        /**
+         * CSS class to be added to the map container while drawing.
+         */
+        drawingCSSClass?: string;
+
+        /**
+         * Layer used to store edit tools (vertex, line guide…).
+         */
+        editLayer?: L.LayerGroup<L.ILayer>;
+
+        /**
+         * Default layer used to store drawn features (marker, polyline…).
+         */
+        featuresLayer?: L.LayerGroup<L.Polyline|L.Polygon|L.Marker>;
+
+        /**
+         * Class to be used as vertex, for path editing.
+         */
+        vertexMarkerClass?: Object;
+
+        /**
+         * Class to be used as middle vertex, pulled by the user to create a new point in the middle of a path.
+         */
+        middleMarkerClass?: Object;
+
+        /**
+         * Class to be used as Polyline editor.
+         */
+        polylineEditorClass?: Object;
+
+        /**
+         * Class to be used as Polygon editor.
+         */
+        polygonEditorClass?: Object;
+
+        /**
+         * Class to be used as Marker editor.
+         */
+        markerEditorClass?: Object;
+
+        /**
+         * Options to be passed to the line guides.
+         */
+        lineGuideOptions?: Object;
+
+        /**
+         * Set this to true if you don't want middle markers.
+         */
+        skipMiddleMarkers?: boolean;
+    }
+
+    /**
+     * Make geometries editable in Leaflet.
+     *
+     * This is not a plug and play UI, and will not. This is a minimal, lightweight, and fully extendable API to
+     * control editing of geometries. So you can easily build your own UI with your own needs and choices.
+     */
+    export interface Editable extends Mixin.LeafletMixinEvents {
+        /**
+         * Options to pass to L.Editable when instanciating.
+         */
+        options: EditOptions;
+
+        currentPolygon: Polyline|Polygon|Marker;
```

Comment:
OK, I suppose it would have been clearer if he named it `currentObject` or something less specific since it's odd to have a `currentPolygon` be a marker or line. :)